plugin/keyring_vault/vault_curl.h

```diff
@@ -18,17 +18,16 @@ namespace keyring
 class Vault_curl : public IVault_curl, private boost::noncopyable
 {
 public:
-  Vault_curl(ILogger *logger)
+  Vault_curl(ILogger *logger, CURL *curl)
     : logger(logger)
-    , curl(NULL)	
+    , curl(curl)
     , list(NULL)
   {}
```

Comment:
curl_errbuf is used in reset_curl_session and get_error_from_curl. If I want to use it as a local variable I would have to declare it locally in all the functions you listed + change the function headers to accept this variable. In other words this is a variable that is used by different functions in the class.
Similar read_data_ss is used in all the function you listed + reset_curl_session
Vault_ca is initialized inside init function and then later used by all the functions (they all call reset_curl_session where it is all used).
List is reused in all the function calls.
Part of the reason why I made those variable class-scope is that they can be reused and safe some time between invocation of each of those functions.